x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -450,9 +470,15 @@ public boolean test(PersistentTasksCustomMetaData.PersistentTask<?> persistentTa
                 return true;
             }
 
-            if (assignment != null && assignment.equals(PersistentTasksCustomMetaData.INITIAL_ASSIGNMENT) == false &&
-                assignment.isAssigned() == false) {
-                // Assignment has failed despite passing our "fast fail" validation
+            if (assignment != null
+                && assignment.equals(PersistentTasksCustomMetaData.INITIAL_ASSIGNMENT) == false
+                && assignment.isAssigned() == false) {
+                assignmentExplanation = assignment.getExplanation();
+                // Assignment failed due to primary shard check.
+                // This is hopefully intermittent and we should allow another assignment attempt.
+                if (assignmentExplanation.contains("not all primary shards are active")) {
```

Comment:
The risk of mistakes in future maintenance would be reduced if the string `"], because not all primary shards are active for the following indices ["` from old line 587 was put in a constant and that same constant was used here too.

Reply:
For sure!